2023 redesign css updates

megoth opened the pull request:
I've rebased the branch on `2023-redesign` and added some CSS classes (.container, .tiles, .button, and added a bit to .legend).

There's more work to do, but I wanted to share my progress.

michielbdejong committed:
Fix some broken links, ref #845

michielbdejong committed:
Broken link about_solid.html

michielbdejong committed:
Fix license link

michielbdejong committed:
Fix two buttons that point nowhere

michielbdejong committed:
Point to Discourse and Matrix, fix https://github.com/solid/team/issues/61

KyraAssaad committed:
Revert "Revert "Add button-purple class""

This reverts commit 6f1667e8165d49da7c95a0eca64bc1ef68bfe340.

megoth committed:
Adding and changing some URLs

Based on feedback from Kyra, which I agree to

michielbdejong committed:
Merge pull request #849 from megoth/fix-some-broken-links

Adding and changing some URLs

michielbdejong committed:
Merge pull request #846 from solid/fix-some-broken-links

Fix some broken links, ref #845

KyraAssaad committed:
Change homepage sections to h2

KyraAssaad committed:
Change font from px to rem; add colour variables

KyraAssaad committed:
Style For Developers page

KyraAssaad committed:
Style page header sections

KyraAssaad committed:
Update text for more consistency

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

csarven committed:
Restructure sectioning for consistency and other minor HTML updates

csarven committed:
Reapply intended styles

megoth committed:
Added some CSS classes

Added .container, .tiles, and .button (and developed .legend a bit more).

Have tried to clean up "old" CSS, i.e. the ones not necessary since I've created some CSS classes, but there is probably more.

Need more work, but wanted to share my progress.

megoth committed:
Merge branch '2023-redesign-css-updates' into 2023-redesign-css-updates

megoth commented:
@KyraAssaad you probably want to review this. The work is not done, but I hope the CSS is a bit easier to maintain with my proposed changes.

VirginiaBalseiro commented:
> Pardon me but I don't understand the fascination with adding random classes to HTML that are essentially corresponding to a particular design in mind. As mentioned elsewhere, HTML really needs to stand on its own so that different CSS can be applied and maintained over time without having to resort to HTML changes 
> 
> All of the classes introduced in this PR: `container`, `container--full`, `container--open-end`, `tiles`, `tiles--very-distinct`, `button`, are 100% about the specific proposed design. They add no structural or semantic "information" to the underlying content such that they'd be meaningful for a different design.
> 
> If classes need be, it is because there is a higher level categorisation of the recurring components that's not particularly expressible.

Agree 100% and we were deliberately trying to avoid this with the initial CSS, albeit with its many issues that we were looking into solving later, and made note of that on the PR.

megoth commented:
Ok, then disregard my suggestions. But I really don't understand how you think this will be maintainable.

megoth committed:
Removed CSS classes that don't contribute semantics

As per feedback from Sarven and Virginia

megoth committed:
Merge branch '2023-redesign-css-updates' of github.com:megoth/solidproject.org into 2023-redesign-css-updates

megoth commented:
@csarven @VirginiaBalseiro I've now removed the CSS classes that didn't contribute any semantics to the HTML. I still think it's a bit painful to maintain, especially for people who scan through the CSS and want to reuse design. I've tried to mitigate this by adding some documentation to those parts in the CSS. I've changed very little in the HTML apart from that, so I hope this looks better.

csarven commented on about.html:
```diff
@@ -105,17 +107,17 @@ <h2 property="schema:name">Features</h2>
                   <p><strong>Example applications</strong></p>
                   <p>Find examples of applications built with Solid.</p>
-                  <a href="" rel="rdfs:seeAlso">See example applications</a>
+                  <a href="https://forum.solidproject.org/c/applications/" rel="rdfs:seeAlso">See example applications</a>
```
```suggestion
                  <a href="https://solidproject.org/apps" rel="rdfs:seeAlso">See example applications</a>
```

What's currently on the website is most comprehensive and up to date. Any particular reason to direct people to the forum?

megoth replied:
This was not part of my update (I only removed the class button).

I agree that https://solidproject.org/apps is (probably) better, but then we should add that page to the new redesign as well.

Maybe @KyraAssaad have some thoughts on this?

csarven commented on about.html:
```diff
@@ -131,7 +133,7 @@ <h2 property="schema:name">Features</h2>
           <li><a href="/for-organizations.html" rel="rdfs:seeAlso">For Organizations</a></li>
           <li><a href="/community.html" rel="rdfs:seeAlso">Community</a></li>
-          <li><a href="/#">License</a></li>
+          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md/#">License</a></li>
```
```suggestion
          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md">License</a></li>
```

We should probably use CC0 or CC-BY for the website content. Perhaps create an issue for consideration out of this?

megoth replied:
I agree, but let's create an issue for it

csarven commented on community.html:
```diff
@@ -37,86 +39,89 @@ <h2 property="schema:name">Where to collaborate</h2>
                   <p><strong>Solid Project Forum</strong></p>
                   <p>Create and respond to comments, questions, and discussions happening in the community.</p>
-                  <a href="" rel="rdfs:seeAlso">Go to Solid Forum</a>
+                  <a href="https://forum.solidproject.org" rel="rdfs:seeAlso">Go to Solid Forum</a>
```
```suggestion
                  <a href="https://forum.solidproject.org/" rel="rdfs:seeAlso">Go to Solid Forum</a>
```

csarven commented on community.html:
```diff
@@ -37,86 +39,89 @@ <h2 property="schema:name">Where to collaborate</h2>
+                            <dt>Organization</dt>
+                            <dd><a href="https://mydata.org/" rel="schema:organizer">MyData</a></dd>
+                            <dt>Date</dt>
+                            <dd><time property="schema:startDate" datatype="xsd:date" content="2023-05-31" datetime="2023-05-31">2023-05-31</time>–<time property="schema:endDate" datatype="xsd:date" content="2023-06-02" datetime="2023-06-02">2023-06-02</time></dd>
```
I know I included `content` in the original but it is not necessary if the date is always going to be yyyy-mm-dd without time. The precision is the same. Value from the text content will be used. If the precision for the machine-readable is desired, and the human gets less precise in text content, then `content` can be handy.

```suggestion
                            <dd><time property="schema:startDate" datatype="xsd:date" datetime="2023-05-31">2023-05-31</time>–<time property="schema:endDate" datatype="xsd:date" datetime="2023-06-02">2023-06-02</time></dd>
```

Not sure if you want to follow this pattern for the rest.

megoth replied:
I have no strong opinions on this, but in general less is better. Do you want me to remove all uses of `content="[date]"` in the HTML?

csarven commented on community.html:
```diff
@@ -130,8 +135,8 @@ <h2 property="schema:name">Events</h2>
           <li><a href="/community.html">Community</a></li>
-          <li><a href="/#">License</a></li>
-          <li><a href="/#">Code of Conduct</a></li>
+          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md/#">License</a></li>
```
```suggestion
          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md">License</a></li>
```

csarven commented on for-developers.html:
```diff
@@ -215,8 +217,8 @@ <h2 property="schema:name">Available Solid Server implementations</h2>
           <li><a href="/community.html">Community</a></li>
-          <li><a href="/#">License</a></li>
-          <li><a href="/#">Code of Conduct</a></li>
+          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md/#">License</a></li>
```
```suggestion
          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md">License</a></li>
```

csarven commented on for-organizations.html:
```diff
@@ -113,8 +115,8 @@ <h2 property="schema:name">Solutions using Solid technology</h2>
           <li><a href="/community.html">Community</a></li>
-          <li><a href="/#">License</a></li>
-          <li><a href="/#">Code of Conduct</a></li>
+          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md/#">License</a></li>
```
```suggestion
          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md">License</a></li>
```

csarven commented on index.html:
```diff
@@ -81,8 +83,8 @@ <h3>Explore the community</h3>
           <li><a href="/community.html">Community</a></li>
-          <li><a href="/#">License</a></li>
-          <li><a href="/#">Code of Conduct</a></li>
+          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md/#">License</a></li>
```
```suggestion
          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md">License</a></li>
```

csarven commented on terms.html:
```diff
@@ -78,8 +78,8 @@ <h2 property="schema:name">URI Persistence Policy</h2>
           <li><a href="/community.html">Community</a></li>
-          <li><a href="/#">License</a></li>
-          <li><a href="/#">Code of Conduct</a></li>
+          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md/#">License</a></li>
```
```suggestion
          <li><a href="https://github.com/solid/solidproject.org/blob/main/LICENSE.md">License</a></li>
```

csarven reviewed (commented):
Using the `header` pattern to isolate some heading and short description makes sense. I wasn't sure to implement this in the original because it wasn't particularly necessary to wrap it. If that idea is what's intended - which should be documented - then lets keep it consistent with header.

megoth committed:
Update about.html

Co-authored-by: Sarven Capadisli <info@csarven.ca>

megoth committed:
Update about.html

Co-authored-by: Sarven Capadisli <info@csarven.ca>

megoth committed:
Update community.html

Co-authored-by: Sarven Capadisli <info@csarven.ca>

megoth committed:
Update community.html

Co-authored-by: Sarven Capadisli <info@csarven.ca>

megoth committed:
Update for-developers.html

Co-authored-by: Sarven Capadisli <info@csarven.ca>

megoth committed:
Update for-organizations.html

Co-authored-by: Sarven Capadisli <info@csarven.ca>

megoth committed:
Update index.html

Co-authored-by: Sarven Capadisli <info@csarven.ca>

megoth committed:
Update terms.html

Co-authored-by: Sarven Capadisli <info@csarven.ca>

megoth commented:
> Using the `header` pattern to isolate some heading and short description makes sense. I wasn't sure to implement this in the original because it wasn't particularly necessary to wrap it. If that idea is what's intended - which should be documented - then lets keep it consistent with header.

Where should it be documented?

megoth commented:
Admittedly, I've skipped terms.html - It currently isn't linked to anywhere, I reckon that should be done?

michielbdejong commented:
Is this still incomplete?

jeswr commented:
Closing as a redesign was done across a range of PRs in September 2025.
